src/Storages/VirtualColumnUtils.cpp

```diff
@@ -226,6 +258,24 @@ void addRequestedFileLikeStorageVirtualsToChunk(
             else
                 chunk.addColumn(virtual_column.type->createColumnConstWithDefaultValue(chunk.getNumRows())->convertToFullColumnIfConst());
         }
+        else if (!hive_map.empty())
+        {
+            bool contains_virtual_column = std::any_of(hive_map.begin(), hive_map.end(),
+                [&](const auto& pair)
+                {
+                    return requested_virtual_columns.contains(pair.first);
+                });
+
+            if (!contains_virtual_column)
+                hive_map.clear(); // If we cannot find any virtual column in requested, we don't add any of them to chunk
```

Comment:
Wait, why do we need this code at all?

Reply:
We are iterating over `requested_virtual_columns`, so we should just check if `hive_map` contains current virtual column name or not.